File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> Pairindex = new HashMap<>();

        for (int i=0; i<nums.length; i++) {
            int num = nums[i];
            if (Pairindex.containsKey(target-num)) {
                return new int[] {
                    i, Pairindex.get(target - num)
                };
            }
            Pairindex.put(num, i);

        }
        return new int[] {};

    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {} # format { num : index } 
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in d:
                return (d[diff], i)
            else:
                d[nums[i]] = i
        print('asdf')
        # return